src/libs/datatableproviders/DataProvider.ts

```diff
@@ -0,0 +1,78 @@
+// define metadata structures
+interface EntityTypeMetadata {
+    attributeNames: string[],
+    count: number,
+    idName: string
+}
+
+export interface EntityMetadata {
+    [index: string]: EntityTypeMetadata
+}
+
+export enum EntityQuerySortDirection {
+    // eslint-disable-next-line no-unused-vars
+    Asc = 'asc',
+    // eslint-disable-next-line no-unused-vars
+    Desc = 'desc'
+}
+
+export enum EntityQueryFilterOperator {
+    // eslint-disable-next-line no-unused-vars
+    And = 'and',
+    // eslint-disable-next-line no-unused-vars
+    Or = 'or'
+}
+
+// define paginated query result structures
+interface EntityQuery {
+    page: number,
+    pageSize: number,
+    sortField: string,
+    sortDirection: EntityQuerySortDirection,
+    filterTerms: string,
+    filterOperator: EntityQueryFilterOperator
+}
+interface EntityQueryResultMetadata {
+    unfilteredCount: number,
+    filteredCount: number,
+    filteredPageCount: number
+}
+interface Entity {
+    name: string,
+    entityType: string,
+    attributes: Record<string, any>
+}
+
+export interface EntityQueryResponse {
+    parameters: EntityQuery,
+    resultMetadata: EntityQueryResultMetadata
+    results: Entity[]
+}
+
+export type GetPage = (_signal: any, _workspaceId: string, _entityType: string, _pageNumber: number,
```

Comment:
What's the significance of the input variable names starting with `_`?

Reply:
It marks those variables as unused, and feels better than `// eslint-disable-next-line no-unused-vars`. Since this is just an empty method definition, they're not used here in this file

Reply:
Replacing `no-unused-vars` with `@typescript-eslint/no-unused-vars` in .eslintrc.js will prevent ESLint from complaining about unused variables in function types.

 https://typescript-eslint.io/rules/no-unused-vars/

Reply:
Also, with so many parameters for these functions, I'd suggest an options object instead of having each parameter listed individually.

https://www.codereadability.com/what-are-javascript-options-objects/